frontend/src/app/order-details/order-details.component.ts

```diff
@@ -35,5 +35,14 @@ export class OrderDetailsComponent implements OnInit {
 
     this.dialog.open(TrackingNumberComponent, dialogConfig);
   }
+  
+  releaseOrder(order : number): void{ 
+    this.orderControllerService.releaseOrder(order).subscribe();
+    window.location.reload();
```

Comment:
Do not reload the page. Itcshould be something like this:   
```
this.orderControllerService.releaseOrder(order).subscribe( () => {
    order?.status = 'NEW';
    order?.assignedTo = undefined;
     order?.lastUpdated = new Date()
});

Reply:
Isn't this implemented in our backend method? Why do it again in front end? And it doesn't solve the problem with reloading. After I release or claim an order the button stays the same and looks like it didn't work until I refresh the page.

Reply:
Yes, it is implemented, but refreshing the whole page is goes against the idea of Single Page Applications. If you don't want to reimplement things, then just refetch the order object like how it is done in the ngOnInit method.